packages/common/src/api/library.ts

```diff
@@ -0,0 +1,74 @@
+import type { full } from '@audius/sdk'
+
+import { createApi } from 'audius-query'
+import { UserCollectionMetadata } from 'models/Collection'
+import { Kind } from 'models/Kind'
+import { makeActivity } from 'services/audius-api-client/ResponseAdapter'
+import { APIActivityV2 } from 'services/audius-api-client/types'
+import { reformatCollection } from 'store/cache/collections/utils/reformatCollection'
+import { encodeHashId } from 'utils/hashIds'
+import { removeNullable } from 'utils/typeUtils'
+
+type GetLibraryAlbumsArgs = {
+  userId: number
+  offset: number
+  limit: number
+  query?: string
+  sortMethod?: full.GetUserLibraryAlbumsSortMethodEnum
+  sortDirection?: full.GetUserLibraryAlbumsSortDirectionEnum
+}
+
+export const libraryApi = createApi({
+  reducerPath: 'libraryApi',
+  endpoints: {
+    getLibraryAlbums: {
+      fetch: async (
+        args: GetLibraryAlbumsArgs,
+        { audiusSdk, audiusBackend }
+      ) => {
+        const {
+          userId,
+          offset,
+          limit,
+          query = '',
+          sortMethod = 'added_date',
+          sortDirection = 'desc'
+        } = args
+        const sdk = await audiusSdk()
+        const { data, signature } =
+          await audiusBackend.signDiscoveryNodeRequest()
+        const { data: rawAlbums = [] } =
+          await sdk.full.users.getUserLibraryAlbums({
+            id: encodeHashId(userId),
+            userId: encodeHashId(userId),
+            offset,
+            limit,
+            query,
+            sortMethod,
+            sortDirection,
+            type: 'all',
+            encodedDataMessage: data,
+            encodedDataSignature: signature
+          })
+        const albumsMetadata = rawAlbums
+          .map((r: APIActivityV2) => makeActivity(r))
+          .filter(removeNullable) as UserCollectionMetadata[]
+        const albums = albumsMetadata.map((am) =>
+          reformatCollection({
```

Comment:
Nice. This makes me wonder if it's worth building in the reformatters for each entity type? Do you know if we always want to transform the data for every instance of collection?